boa/src/builtins/number/mod.rs

```diff
@@ -27,11 +24,16 @@ use crate::{
 };
 use num_traits::float::FloatCore;
 
+mod conversions;
+
+#[cfg(test)]
+mod tests;
+
 const BUF_SIZE: usize = 2200;
 
 /// `Number` implementation.
 #[derive(Debug, Clone, Copy)]
-pub(crate) struct Number;
+pub(crate) struct Number(f64);
```

Comment:
Do you think we could/should have a fast path for integers here?

Reply:
> Do you think we could/should have a fast path for integers here?

If you mean to make the Number Object have a integer and rational, I don't think this would be beneficial since every number operation converts the object to a primitive, it is not like we are doing with primitives, and it would complicate the code.

I have some progress locally were I made a fast path for when the value is a `Value::Integer` for `to_int32` and `to_uint32`.